### PR TITLE
Added top_a and repetition_penalty samplers.

### PR DESCRIPTION
**Pull request by 50h100a**
top_a and top_p are applied simultaneously in this implementation.

The repetition_penalty is a variant which does not make penalized tokens with negative logits *more* likely.

### CODE REVIEW

**Comment by AlpinDale** [comment]
Good work, will merge this after #23

File: aphrodite/modeling/layers/sampler.py [hunk 1]
```diff
@@ -241,24 +254,28 @@ def _get_temperatures(input_metadata: InputMetadata) -> List[float]:
...
+
+        top_ps += [sampling_params.top_p] * len(seq_ids)
         top_ks += [top_k] * len(seq_ids)
-    return top_ps, top_ks
+        top_as += [sampling_params.top_a] * len(seq_ids)
+
+    return top_ps, top_ks, top_as
 
 
+def _apply_top_ap_top_k(
```
Comment (StefanGliga) [inline aphrodite/modeling/layers/sampler.py:278]: looks like something got messed up here

**Review by StefanGliga — CHANGES_REQUESTED** [review]

File: aphrodite/modeling/layers/sampler.py (hunk 1 above)
Comment (50h100a) [reply aphrodite/modeling/layers/sampler.py:278]: No typo, it represents that the Top-A and Top-P samplers were merged into a single pass.
Possibly confusing, but easy to fix.

Comment (StefanGliga) [reply aphrodite/modeling/layers/sampler.py]: lgtm
